configuration/passport.js

```diff
@@ -0,0 +1,38 @@
+const passport = require( 'passport' )
+const LocalStrategy = require( 'passport-local' ).Strategy
+
+const User = require( '../database/user' )
+
+const strategy = new LocalStrategy({
+    usernameField: 'email',
+    session: false,
+    passReqToCallback: true
+}, ( request, email, password, done ) => {
+  User.findByEmail( email )
+    .then( user => {
+      if( User.validPassword( password,  user.password ) ) {
+        return done( null, user )
+      } else {
+        return done( null, false, request.flash( 'message', 'Oops! Wrong password.' ))
+      }
+    })
+    .catch( error => {
+      return done( null, false, request.flash( 'message', 'Seems we don\'t have your email...' ) )
```

Comment:
What if the error is not a "missing email"? How are you confirming this?

To handle these errors more specifically, you should check the `error` object and only respond with the "we don't have your email" if that is _actually_ what the error is.

Otherwise, it should log the error and show an error page.

The danger with the way you've currently written error handling is that you are setting the app up to "swallow" errors and potentially give incorrect messages to the user.